DigitalLearningSolutions.Web/Scripts/trackingSystem/centreCourseSetup.ts

```diff
@@ -0,0 +1,54 @@
+const copyCourseLinkClass = 'copy-course-link';
+const copyLinkIdPrefix = 'copy-course-';
+const launchCourseButtonIdPrefix = 'launch-course-';
+
+setupCourseLinkClipboardCopiers();
+
+function setupCourseLinkClipboardCopiers() {
+  const copyCourseLinks = document.getElementsByClassName(copyCourseLinkClass);
+
+  for (let i = 0; i < copyCourseLinks.length; i += 1) {
+    const currentLink = copyCourseLinks[i];
+    const linkId = currentLink.id;
+    const customisationId = linkId.slice(copyLinkIdPrefix.length);
+
+    currentLink.addEventListener('click', (e) => {
+      e.preventDefault();
+      e.stopPropagation();
+      copyLaunchCourseLinkToClipboard(customisationId);
+    });
+  }
+}
+
+function copyLaunchCourseLinkToClipboard(customisationId: string) {
+  const launchCourseButtonId = launchCourseButtonIdPrefix + customisationId;
+  const copyText = document.getElementById(launchCourseButtonId) as HTMLAnchorElement;
+  const link = copyText.href;
+  const succeeded = copyTextToClipboard(link);
+  let alertMessage: string;
+
+  if (succeeded) {
+    alertMessage = `Copied the text: ${link}`;
+  } else {
+    alertMessage = `Copy not supported or blocked. Try manually selecting and copying: ${link}`;
+  }
+
+  alert(alertMessage);

```

Comment:
Alerts were used on the old site as confirmation that the copying was either completed successfully or was blocked in some form. It seems pretty convenient to me, so I've introduced it here. A quick run through with the screen reader seems okay to me, but I'm happy to change this if we have a preference for an alternative method.

Reply:
It was decided to keep the alert as is at the Show & Tell on 16th July 2021. The current system uses this, and they have received no complaints about it, so it shall do.